org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/preferences/formatter/ModifyDialogTabPage.java

```diff
@@ -862,8 +863,85 @@ public void controlResized(ControlEvent e) {
 
 		Label sashHandle = new Label(scrollContainer, SWT.SEPARATOR | SWT.VERTICAL);
 		gridData= new GridData(SWT.RIGHT, SWT.FILL, false, true);
+		gridData.verticalSpan= 2;
 		sashHandle.setLayoutData(gridData);
 
+		Composite buttonComposite= new Composite(scrollContainer, SWT.NONE);
+		gridData= new GridData(SWT.FILL, SWT.END, true, false);
+		buttonComposite.setLayoutData(gridData);
+		layout= new GridLayout(1, false);
+		layout.verticalSpacing= (int)(1.5 * fPixelConverter.convertVerticalDLUsToPixels(IDialogConstants.VERTICAL_SPACING));
+		layout.horizontalSpacing= fPixelConverter.convertHorizontalDLUsToPixels(IDialogConstants.HORIZONTAL_SPACING);
+		layout.marginHeight= fPixelConverter.convertVerticalDLUsToPixels(IDialogConstants.VERTICAL_MARGIN);
+		layout.marginWidth= fPixelConverter.convertHorizontalDLUsToPixels(IDialogConstants.HORIZONTAL_MARGIN);
+		buttonComposite.setLayout(layout);
+
+		Group buttonHolder= new Group(buttonComposite, SWT.NONE);
+		gridData= new GridData(SWT.FILL, SWT.END, true, false);
+		buttonHolder.setLayoutData(gridData);
+		layout= new GridLayout(4, false);
+		layout.verticalSpacing= (int)(1.5 * fPixelConverter.convertVerticalDLUsToPixels(IDialogConstants.VERTICAL_SPACING));
+		layout.horizontalSpacing= fPixelConverter.convertHorizontalDLUsToPixels(IDialogConstants.HORIZONTAL_SPACING);
+		layout.marginHeight= fPixelConverter.convertVerticalDLUsToPixels(IDialogConstants.VERTICAL_MARGIN);
+		layout.marginWidth= fPixelConverter.convertHorizontalDLUsToPixels(IDialogConstants.HORIZONTAL_MARGIN);
+		buttonHolder.setLayout(layout);
+
+		Button b= new Button(buttonHolder, SWT.PUSH);
+		b.setText(FormatterMessages.ModifyDialogTabPage_selectAll_text);
+		gridData= new GridData(SWT.LEFT, SWT.TOP, false, true);
+		b.setLayoutData(gridData);
+		b.setEnabled(true);
+		b.setVisible(true);
+		SWTUtil.setButtonDimensionHint(b);
+		b.addSelectionListener(new SelectionAdapter(){
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				doSetAll(true);
+			}
+		});
+
+		b= new Button(buttonHolder, SWT.PUSH);
+		b.setText(FormatterMessages.ModifyDialogTabPage_deselectAll_text);
+		gridData= new GridData(SWT.LEFT, SWT.TOP, false, true);
+		b.setLayoutData(gridData);
+		b.setEnabled(true);
+		b.setVisible(true);
+		SWTUtil.setButtonDimensionHint(b);
+		b.addSelectionListener(new SelectionAdapter(){
```

Comment:
Using `SelectionListener.widgetSelectedAdapter()` would allow you to use a lambda and therefore would make the code more compact.
This could also be applied to the following two occurrences.

Reply:
Not a fan of lambdas, but sure, done.